### PR TITLE
Debug tools feature

### PR DESCRIPTION
**Pull request by lo1ol**
Feature with debug utilities for initrd image

### CODE REVIEW

File: features/debug-tools/config.mk [hunk 1]
```diff
@@ -1,2 +1,2 @@
 DEBUG_TOOLS_PROGS := gdb strace
-DEBUG_TOOLS_FILES := $(DATADIR)/gdb
+DEBUG_TOOLS_FILES := $(DATADIR)/gdb $(LIBDIR)/terminfo/ $(DATADIR)/terminfo/
```
Comment (legionus) [inline features/debug-tools/config.mk:2]: Another problem is that only on systemd-based installations `/lib` and `/usr/lib` are the same thing.

```
ls -1d {,/usr}/{share,lib{,64}}/terminfo
ls: cannot access '/share/terminfo': No such file or directory
ls: cannot access '/lib64/terminfo': No such file or directory
ls: cannot access '/usr/lib/terminfo': No such file or directory
ls: cannot access '/usr/lib64/terminfo': No such file or directory
/lib/terminfo
/usr/share/terminfo
```
You do not need to add `$(LIBDIR)/terminfo/` at all. The terminfo database has symlinks in / lib if needed.

File: features/debug-tools/rules.mk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+GDB_PYTHON_VERSION=$(shell $(FEATURESDIR)/debug-tools/bin/get-gdb-python-version gdb)
+
+PUT_FEATURE_PROGS += $(DEBUG_TOOLS_PROGS)
+PUT_FEATURE_FILES += $(DEBUG_TOOLS_FILES) $(LIBDIR)/python$(GDB_PYTHON_VERSION)
```
Comment (legionus) [inline features/debug-tools/rules.mk]: are you sure about `$(LIBDIR)/python$(GDB_PYTHON_VERSION)` ?
It looks like you want to pack `sys.path`.

File: features/debug-tools/bin/get-gdb-python-version [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+gdb=$1
+
+echo -e "py\nimport sys\nprint (sys.version)\nend\n quit" | ${gdb} -q |  head -n 1 |grep -o "[0-9]\+\.[0-9]\+"
```
[the remark this answers is not in the document]
Comment (lo1ol) [reply features/debug-tools/bin/get-gdb-python-version]: fixed

File: features/debug-tools/config.mk (hunk 1 above)
Comment (lo1ol) [reply features/debug-tools/config.mk]: fixed

[the remark this answers is not in the document]
Comment (lo1ol) [reply features/debug-tools/config.mk]: fixed

File: features/debug-tools/rules.mk [hunk 4]
```diff
@@ -0,0 +1,10 @@
+ifeq "$(GDB_PROG)" ''
+GDB_PROG = gdb
+endif
+
+ifneq '$(shell ldd "$(shell which $(GDB_PROG))" | grep libpython)' ''
+GDB_PYTHON_PATHES = $(shell $(GDB_PROG) -q --ex 'python print(" ".join(sys.path))' --ex 'quit')
```
Comment (legionus) [inline features/debug-tools/rules.mk:6]:
```
s/GDB_PYTHON_PATHES/GDB_PYTHON_PATHS/
```

Comment (lo1ol) [reply features/debug-tools/rules.mk]: fixed

**Pull request merged**